kani-compiler/src/kani_middle/reachability.rs

```diff
@@ -13,146 +13,135 @@
-pub fn filter_const_crate_items<F>(tcx: TyCtxt, mut predicate: F) -> Vec<MonoItem>
+pub fn filter_const_crate_items<F>(tcx: TyCtxt, mut predicate: F) -> Vec<InternalMonoItem>
 where
     F: FnMut(TyCtxt, DefId) -> bool,
 {
-    let mut roots = Vec::new();
-    for hir_id in tcx.hir_crate_items(()).items() {
-        let def_id = hir_id.owner_id.def_id.to_def_id();
-        let def_kind = tcx.def_kind(def_id);
-        if matches!(def_kind, DefKind::Const) && predicate(tcx, def_id) {
-            let instance = Instance::mono(tcx, def_id);
-            let body = tcx.instance_mir(InstanceDef::Item(def_id));
-            let mut collector =
-                MonoItemsFnCollector { tcx, body, instance, collected: FxHashSet::default() };
-            collector.visit_body(body);
-
-            roots.extend(collector.collected);
-        }
-    }
-    roots
-}
-
-fn is_generic(tcx: TyCtxt, def_id: DefId) -> bool {
-    let generics = tcx.generics_of(def_id);
-    generics.requires_monomorphization(tcx)
-}
-
-fn to_mono_root(tcx: TyCtxt, item_id: ItemId, def_id: DefId) -> Option<MonoItem> {
-    let kind = tcx.def_kind(def_id);
-    match kind {
-        DefKind::Static(..) => Some(MonoItem::Static(def_id)),
-        DefKind::Fn => Some(MonoItem::Fn(Instance::mono(tcx, def_id))),
-        DefKind::GlobalAsm => Some(MonoItem::GlobalAsm(item_id)),
-        _ => {
-            debug!(?def_id, ?kind, "Ignored item. Not a root type.");
-            None
+    rustc_smir::rustc_internal::run(tcx, || {
+        let crate_items = stable_mir::all_local_items();
+        let mut roots = Vec::new();
+        // Filter regular items.
+        for item in crate_items {
+            // Only collect monomorphic items.
+            if let Ok(instance) = Instance::try_from(item) {
+                let def_id = rustc_internal::internal(&item);
+                if predicate(tcx, def_id) {
+                    let body = instance.body().unwrap();
+                    let mut collector = MonoItemsFnCollector {
+                        tcx,
+                        body: &body,
+                        collected: FxHashSet::default(),
+                        instance: &instance,
+                    };
+                    collector.visit_body(&body);
+                    roots.extend(collector.collected.iter().map(rustc_internal::internal));
+                }
+            }
         }
-    }
+        roots
+    })
+    .unwrap()
 }
 
 struct MonoItemsCollector<'tcx> {
     /// The compiler context.
     tcx: TyCtxt<'tcx>,
     /// Set of collected items used to avoid entering recursion loops.
-    collected: FxHashSet<MonoItem<'tcx>>,
+    collected: FxHashSet<MonoItem>,
     /// Items enqueued for visiting.
-    queue: Vec<MonoItem<'tcx>>,
+    queue: Vec<MonoItem>,
     #[cfg(debug_assertions)]
-    call_graph: debug::CallGraph<'tcx>,
+    call_graph: debug::CallGraph,
 }
 
+// TODO: Need to figure out how to handle stubbing errors.
```

Comment:
What's this referring to? Please open an issue (unless we have it) and refer to it here.